Tests/SwiftDocCTests/Infrastructure/PathHierarchyTests.swift

```diff
@@ -3126,7 +3126,103 @@ class PathHierarchyTests: XCTestCase {
+                ]
+            )),
+            JSONFile(name: "MainModule@ImportedModule.symbols.json", content: makeSymbolGraph(
+                moduleName: "MainModule",
+                symbols: [
+                    makeSymbol(id: importedProtocolID, kind: .protocol, pathComponents: ["BaseProtocol"]),
+                    makeSymbol(id: importedTypeID, kind: .struct, pathComponents: ["ExtendedType"]),
+                    makeSymbol(
+                        id: extensionSymbolID,
+                        kind: .extension,
+                        pathComponents: ["ExtendedType"],
+                        otherMixins: [extensionMixin]
+                    ),
+                    makeSymbol(
+                        id: extensionMethodID,
+                        kind: .method,
+                        pathComponents: ["ExtendedType", "extensionMethod()"],
+                        otherMixins: [extensionMixin]
+                    )
+                ],
+                relationships: [
+                    .init(
+                        source: extensionMethodID,
+                        target: extensionSymbolID,
+                        kind: .memberOf,
+                        targetFallback: "ImportedModule.ExtendedType"
+                    ),
+                    .init(
+                        source: extensionSymbolID,
+                        target: importedTypeID,
+                        kind: .extensionTo,
+                        targetFallback: "ImportedModule.ExtendedType"
+                    )
+                ]
+            ))
+        ])
+
+        let (_, context) = try await loadBundle(catalog: catalog)
+        let tree = context.linkResolver.localResolver.pathHierarchy
+
+        XCTAssertEqual(tree.modules.count, 1)
+        XCTAssertEqual(tree.modules.first?.name, "MainModule")
+
+        let paths = tree.caseInsensitiveDisambiguatedPaths()
+        XCTAssertEqual(paths[importedProtocolID], "/MainModule/ImportedModule/BaseProtocol")
+        XCTAssertEqual(paths[importedTypeID], "/MainModule/ImportedModule/ExtendedType-struct")
+        XCTAssertEqual(
+            paths[extensionMethodID],
+            "/MainModule/ImportedModule/ExtendedType/extensionMethod()"
+        )
+
+        // Verify that symbols can be found at their correct paths
+        try assertFindsPath("/MainModule/ImportedModule/BaseProtocol", in: tree, asSymbolID: importedProtocolID)
+        try assertFindsPath("/MainModule/ImportedModule/ExtendedType-struct", in: tree, asSymbolID: importedTypeID)
+        try assertFindsPath(
+            "/MainModule/ImportedModule/ExtendedType/extensionMethod()",
+            in: tree,
+            asSymbolID: extensionMethodID
+        )
+
+        // Verify that absolute paths to non-existent modules throw moduleNotFound error
+        // This is the fix being tested: without it, single-module fallback would trigger incorrectly
```

Comment:
minor: What this comment is describing and what the test assertions below are meant to verify isn't really the same thing. 

If we only wanted to verify that non-existent modules result in a moduleNotFound error, then we could use a more explicit path like `"/NotFound"`.

What the test assertions below are meant to verify is that a link that resolves relative to the module fails to resolve as an absolute link, with a moduleNotFound error.

It would be good to both verify that `"ExtendedModule/ExtendedType/extensionMethod()"` resolves _and_ that `"/ExtendedModule/ExtendedType/extensionMethod()"` (with a leading slash) raises the expected error.

Reply:
Also, if it's only these test assertions that are important, then the other test assertions could be removed along with that portion of the code comment.